components/datetime/src/format/time_zone.rs

```diff
@@ -51,8 +52,83 @@ where
     T: TimeZoneInput,
     W: fmt::Write + ?Sized,
 {
-    let pattern = &time_zone_format
-        .patterns
+    match (&time_zone_format.patterns, &time_zone_format.config) {
+        (Some(patterns), None) => write_pattern(time_zone_format, time_zone, patterns, w),
+        (None, Some(config)) => write_config(time_zone_format, time_zone, config, w),
+        (Some(_), Some(_)) => {
+            unreachable!("TimeZoneFormat should have a configuration or a pattern, but found both")
+        }
+        (None, None) => unreachable!(
+            "TimeZoneFormat should have either a configuration or a pattern, but found neither"
+        ),
+    }
+}
+
+fn write_config<T, W>(
+    time_zone_format: &TimeZoneFormat,
+    time_zone: &T,
+    config: &TimeZoneFormatConfig,
```

Comment:
> nitpick

Since this is a `Copy` type we should just take it by value, instead of by reference.

Reply:
I was using `TimeZoneFormatConfig` at the beginning, but it turned out that config is type `&TimeZoneFormatConfig`. If we want to use `config: TimeZoneFormatConfig`, should I use `(None, Some(config)) => write_config(time_zone_format, time_zone, *config, w)` in `write_zone` method above?

Reply:
@samchen61661 

Just remove the reference from the top-level match

```diff
-    match (&time_zone_format.patterns, &time_zone_format.config) {
+    match (&time_zone_format.patterns, time_zone_format.config) {
```

And then you should be able to take everything else by value, including `IsoMinutes` etc. no problem.

Reply:
Thanks Eric! It works!